File: backend/src/main/java/com/finchsolucoes/testejavafinchsolucoes/model/Ingrediente.java
package com.finchsolucoes.testejavafinchsolucoes.model;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.PositiveOrZero;

import org.hibernate.validator.constraints.Length;

@Entity
public class Ingrediente implements Serializable{
	private static final long serialVersionUID = 1L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@NotEmpty(message = "O campo nome é obrigatório!")
	@Length(min = 3, max = 80, message = "O campo nome deve ter entre 3 e 80 caracteres!")
	private String nome;
	
	@PositiveOrZero(message = "O preço não pode ser negativo!")
	private Double preco;
	
	public Ingrediente() {}
	
	public Ingrediente(Integer id, String nome, Double preco) {
		super();
		this.id = id;
		this.nome = nome;
		this.preco = preco;
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Double getPreco() {
		return preco;
	}

	public void setPreco(Double preco) {
		this.preco = preco;
	}	
}

File: web/scripts/model/ingredientes/Ingrediente.js
class Ingrediente{
    constructor(id, nome, preco){
        this._id = id;
        this._nome = nome;
        this._preco = preco;
    }
    set id(id){
        this._id = id;
    }
    get id(){
        return this._id;
    }
    set nome(nome){
        this._nome = nome;
    }
    get nome(){
        return this._nome;
    }
    set preco(preco){
        this._preco = preco;
    }
    get preco(){
        return this._preco;
    }
}